model/countries/AU/AU-parsing-rules.yaml

```diff
@@ -0,0 +1,376 @@
+regex_definitions:
+  kBuildingValueRe:  # Regex for type "building"
+    regex_concat:
+      parts:
+      - regex_fragment: \d+  # House number
+      # Covering cases - house number with alphabetic suffix, house number range.
+      - regex_fragment: '(?:'
+      - regex_fragment: \s*[[:alpha:]]\b # Optional suffix for building ids with a letter suffix, e.g. 12a.
+      - regex_fragment: '|'
+      - regex_fragment: \s*?[-]\s*\d+ # Housnumber range (10-17)
+      - regex_fragment: ')?'
+      wrap_non_capture: false
+
+  kUnitTypeLiteralRe:  # Regex for "unit-type"
+    regex_fragment: (?:\b(?:apt\.?|apartment|se\.?|ste\.?|suite|unit|u\.?|room|rm\.?|shop|po box|ms\.?|mail shop|mb\.?|marine berth|vlla\.?|villa|we\.?|warehouse|office|off\.?|fy\.?|factory|sl\.?|stall|site|shed))
+
+  kUnitNameValueRe:  # Regex for "unit-name"
+    regex_fragment: (?:\d+\w?\b|\w\b)
+
+  kFloorLiteralRe:  # Regex for floor literal
+    regex_fragment: (?:\b(?:level|l\.?|floor|fl\.?|basement|b\.?|ground floor|g\.?|lower ground floor|lg\.?|upper ground floor|ug\.?|mezzanine|m\.?))
```

Comment:
As discussed `basement|b\.?|ground floor|g\.?|lower ground floor|lg\.?|upper ground floor|ug\.?|mezzanine|m\.?` probably don't belong into `kFloorLiteralRe`. "basement" is not a prefix to a number but a value of the floor.

Reply:
As discussed in private, refactored everything and resolved the issue having with Parsing floor. Thanks!